### PR TITLE
Update build configuration to make it multi-platform friendly

### PR DESCRIPTION
**Pull request by tolbrino**
Ref #1771 

This work is the baseline for the Windows build support. The changes are
not specific to Windows, but are required to make the project compile in
a Windows environment.

### CODE REVIEW

**Comment by tolbrino** [comment]
@ThomasArts Does this build and test properly for you locally? I can't run the tests yet myself.

**Comment by lucafavatella** [comment]
I moved WIP from the title to the label so to ease ticket filtering.

File: .circleci/config.yml [hunk 1]
```diff
@@ -281,7 +281,7 @@ jobs:
       - *restore_rebar_cache
       - run:
           name: Build
-          command: ./rebar3 as test do release
```
Comment (hanssv) [inline .circleci/config.yml]: This does not seem to be the same thing?!

Comment (tolbrino) [reply .circleci/config.yml]: Fixed.

File: Makefile [hunk 2]
```diff
@@ -158,23 +167,28 @@ dev3-clean: internal-clean
...
 	if [ $$(printf "%b" "$${EPOCH_PROCESSES}" | wc -l) -gt 0 ] ; then \
 		(printf "%b\n%b\n" "Failed testing: another Epoch node is already running" "$${EPOCH_PROCESSES}" >&2; exit 1);\
 	else \
-		./rebar3 as test do release, ct $(CT_TEST_FLAGS) --sys_config config/test.config; \
+		$(REBAR) ct $(CT_TEST_FLAGS) --sys_config config/test.config; \
 	fi
 
-eunit:
-	@ERL_FLAGS="-args_file $(EUNIT_VM_ARGS)" ./rebar3 do eunit $(EUNIT_TEST_FLAGS)
+REVISION:
```
Comment (dincho) [inline Makefile]: this won't work once you change the HEAD

Comment (dincho) [reply Makefile]: I mean, it will keep the old REVISION file

Comment (tolbrino) [reply Makefile]: Its `phony`, so it should be re-executed all the time.

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,8 @@
 CORE = rel/epoch/bin/epoch
 VER := $(shell cat VERSION)
 
+REBAR ?= ./rebar3
```
Comment (uwiger) [inline Makefile]: Would it perhaps be even better with
`REBAR ?= $(shell which rebar3 || echo ./rebar3)`?

(Though I don't know how that works on Windows...)

File: rebar.config [hunk 4]
```diff
@@ -24,29 +24,27 @@
...
                       {ref, "69b4ada"}}}, % tag: 3.6.7
-        {setup, "2.0.2"},
-        {parse_trans, "3.2.0"},
         {mnesia_rocksdb, ".*", {git, "https://github.com/aeternity/mnesia_rocksdb.git", {ref,"ad8e7b6"}}},
         {enacl, ".*", {git, "https://github.com/jlouis/enacl.git", {ref,"885662c"}}},
         {enoise, "1.0.0"},
         {cowboy, "2.2.2", {git, "https://github.com/ninenines/cowboy.git", {tag, "2.2.2"}}},
-        {jsx, {git, "https://github.com/talentdeficit/jsx.git", {tag, "2.8.0"}}},
-        {jesse, {git, "https://github.com/aeternity/jesse.git", {ref, "4640ad1"}}}, %% Tag `1.5.0-rc2` with workaround for unexported `http_uri:uri()` type.
+        {jesse, {git, "https://github.com/tolbrino/jesse.git",
```
Comment (uwiger) [inline rebar.config]: Shouldn't this be merged into the `aeternity` fork?

File: rebar.config [hunk 5]
```diff
@@ -24,29 +24,27 @@
...
         {base58, {git, "https://github.com/aeternity/erl-base58.git", {ref,"60a3356"}}},
-        {eper, ".*", {git, "git://github.com/aeternity/eper.git", {ref, "03d4be3"}}},
+        {eper, "0.99.1"},
         {erlexec, ".*", {git, "https://github.com/saleyn/erlexec.git", {ref, "97a5188"}}},
         {sha3, {git, "https://github.com/szktty/erlang-sha3", {ref, "dbdfd12"}}},
         {sext, {git, "https://github.com/uwiger/sext.git", {ref, "615eebc"}}},
         {idna, {git, "https://github.com/aeternity/erlang-idna", {tag, "6.0.0"}}}
        ]}.
 
 %% pc needed for erlexec and version 1.6.0 works, but 1.9.1 does not.
```
Comment (dincho) [inline rebar.config]: As the plugin is not used anymore (?) I guess you could remove this commend as well

Comment (tolbrino) [reply rebar.config]: Done.

**Pull request merged**